external_systems/sources/_proxies.py

```diff
@@ -26,12 +26,24 @@ class CustomCaBundleSession(Session):
     A wrapper for requests.Session to override 'verify' property, ignoring REQUESTS_CA_BUNDLE environment variable.
 
     This is a workaround for https://github.com/psf/requests/issues/3829 (will be fixed in requests 3.0.0)
+    
+    The standard behavior of requests is to ALWAYS use the REQUESTS_CA_BUNDLE environment variable here if "verify" is not set on
+    the request (even if it's set on the session level).
     """
 
     def merge_environment_settings(self, url, proxies, stream, verify, *args, **kwargs):  # type: ignore[no-untyped-def]
-        if isinstance(self.verify, str) and os.path.exists(self.verify):
+        user_has_manually_overridden_verify = verify is not None
```

Comment:
verify defaults to None: https://github.com/psf/requests/blob/c65c780849563c891f35ffc98d3198b71011c012/src/requests/sessions.py#L515